remo/profiles/forms.py

```diff
@@ -137,6 +137,16 @@ def clean_twitter_account(self):
         twitter_account = self.cleaned_data['twitter_account']
         return twitter_account.strip('@')
 
+    def clean_mentor(self):
+        """Make sure that we don't save an empty mentor"""
+        has_perm_to_edit_mentor = self.request.user.has_perm('profiles.can_change_mentor')
+        cleaned_mentor = self.cleaned_data['mentor']
+        if has_perm_to_edit_mentor == False and cleaned_mentor == None:
+            initial_mentor = self.initial['mentor']
+            cleaned_mentor = User.objects.get(id=initial_mentor)
+        return cleaned_mentor
```

Comment:
There is no need to query the database to get the initial mentor. You could write something like this:

```python
if not self.request.user.has_perm('profiles.can_change_mentor'):
    self.cleaned_data['mentor'] = self.instance.mentor
return self.cleaned_data['mentor']
```